torchaudio/datasets/librispeech.py

```diff
@@ -76,7 +76,26 @@ def load_librispeech_item(
     )
 
 
-class LIBRISPEECH(Dataset):
+def load_librispeech_item(
+    fileid: str, path: str, ext_audio: str, ext_txt: str
+) -> Tuple[Tensor, int, str, int, int, int]:
+    filepath, sample_rate, transcript, speaker_id, chapter_id, utterance_id = get_librispeech_metadata(
+        fileid, path, ext_audio, ext_txt
+    )
+    full_path = os.path.join(path, filepath)
+    waveform, _ = torchaudio.load(full_path)
+
+    return (
+        waveform,
+        sample_rate,
+        transcript,
+        speaker_id,
+        chapter_id,
+        utterance_id,
+    )
+
+
+class LIBRISPEECHBase(Dataset):
```

Comment:
Let's not keep dragging the bad naming scheme.

```suggestion
class LibriSpeechBase(Dataset):
```

Reply:
Also, since we are introducing a new class, which gives rare opportunity to redefine the interface without BC, let's remove `folder_in_archive`, and make `root` optional.

Reply:
Similarly, we should split download from the constructor, downloading should not be the responsibility of this class (or at least constructor).

Reply:
sure, we can redefine/simplify the API for the base class. curious what's the reasoning for removing downloading from this class?

Reply:
Downloading is a complex procedure. It involves network access, local file access and the permission.
There are so many ways to fail for downloading feature, for example, network unreachable, slow network, directory creation permission, file write permission, and file storage limit. The fact that the existing dataset interface ended up having `folder_in_archive` argument is one testament.

Also the data integrity is not resolved with the current approach. It is easy to verify the integrity of downloaded archive by using SHA-256 hash, but what about the individual data? Say, I delete one of the WAV file from the directory, should the download function refill it somehow? How to detect it? All these specifications and responsibility of flawless executions should be placed in a separate location. Dataset class should not be responsible. 

From the perspective of library aiming to be building block (or single responsibility principle), it's not something that should be performed inside of dataset constructor. It should be performed separately.
By introducing the Base dataset class, we can re-interpret the existing dataset classes as higher-level implementations, composed of multiple primitives.